File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize(*multiples)
    @multiples = multiples
  end

  def to(limit)
    (1...limit).reduce(0) do |sum, num|
      sum += num if multiples.any? { |multiple| is_multiple?(num, multiple) }
      sum
    end
  end

  private

  attr_reader :multiples

  def is_multiple?(num, multiple)
    (num % multiple).zero?
  end
end

File: java/sum-of-multiples/src/main/java/SumOfMultiples.java
import java.util.stream.IntStream;

public final class SumOfMultiples {

    private final int maxValue;
    private final int[] multiples;

    public SumOfMultiples(int maxValue, int[] multiples) {
        this.maxValue = maxValue;
        this.multiples = multiples;
    }

    public int getSum() {
        return IntStream
                .range(1, maxValue)
                .filter(n -> isMultiple(n))
                .sum();
    }

    private Boolean isMultiple(int value) {
        for (int multiple : multiples) {
            if (value % multiple == 0) {
                return true;
            }
        }
        return false;
    }
}